src/main.cpp

```diff
@@ -1834,8 +1834,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         return true;
     }
 
-    bool fScriptChecks = true;
-    if (fCheckpointsEnabled) {
+    assert(nScriptCheckThreads >= 1);
```

Comment:
I think running with -par=1 sets nScriptCheckThreads to zero, to mean "no concurrency"
(see comment on line 931 of init.cpp)